Abracadabra/AbracadabraAPI/AbracadabraAPI/Services/UpdateTrendingService.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using AbracadabraAPI.Data;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace AbracadabraAPI.Services
+{
+    public class UpdateTrendingService : IHostedService, IDisposable
+    {
+        private readonly long _originalTime = 1577883600;
+
+        private readonly ILogger<UpdateTrendingService> _logger;
+        private readonly IServiceScopeFactory _scopeFactory;
+        private Timer _timer;
+
+        public UpdateTrendingService(ILogger<UpdateTrendingService> logger, IServiceScopeFactory scopeFactory)
+        {
+            _logger = logger;
+            _scopeFactory = scopeFactory;
+        }
+
+        public Task StartAsync(CancellationToken cancellation)
+        {
+            _logger.LogInformation("Update trending service is running.");
+
+            _timer = new Timer(UpdateTrending, null, TimeSpan.Zero,
+                TimeSpan.FromMinutes(1));
+
+            return Task.CompletedTask;
+        }
+
+        private void UpdateTrending(object state)
+        {
+            using (var scope = _scopeFactory.CreateScope())
+            {
+                var context = scope.ServiceProvider.GetRequiredService<AbracadabraContext>();
+                foreach (var question in context.Questions)
+                {
+                    var score = question.Upvotes - question.Downvotes;
+
+                    var order = Math.Log10(Math.Max(Math.Abs(score), 1));
+
+                    //double sign1 = Math.Sign(order);
```

Comment:
Can it be automated using Math.Sign instead of the if statements?

Reply:
Updated in the latest commit.